data/publiccloud/restart_instance.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -e
+
+check_ssh()
+{
+    nc -w1 -z $host 22
+}
+
+wait_for_power_off()
+{
+    local host=$1
+    local tries=$2
+    while check_ssh $host
+    do
+        tries=$(( $tries - 1  ))
+        [ $tries -lt 1 ] && return 1;
+        echo "waiting for power off"
+        sleep 1;
+    done
+    return 0;
+}
+
+wait_for_power_on()
+{
+    local host=$1
+    local tries=$2
+    while ! check_ssh $host
+    do
+        tries=$(( $tries - 1  ))
+        [ $tries -lt 1 ] && return 1;
+        echo "waiting for power on"
+        sleep 1;
+    done
+    return 0;
+}
+
+if [ $# -lt 3 ]; then
+    echo  "$0 (EC2|AZURE) <instance_id> <host>"
+    exit 2;
+fi
+
+PROVIDER=$1
+INSTANCE_ID=$2
+HOST=$3
+CNT=60;
```

Comment:
is 60 sec enough to wait for the VM to be off or on?

Reply:
I increased it to 120. So we have 2min for power off and 2min fro power on